biojava-structure/src/main/java/org/biojava/nbio/structure/io/mmcif/DownloadChemCompProvider.java

```diff
@@ -337,6 +345,12 @@ private static  boolean fileExists(String recordName){
 
 		File f = new File(fileName);
 
+		// delete files that are too short to have contents
+		if( f.length() < LocalPDBDirectory.MIN_PDB_FILE_SIZE ) {
+			f.delete();
+			return false;
```

Comment:
If the file fails to be deleted, this won't work as expected. I think the fail to delete case should throw an IOException

Reply:
I don't check the status because the delete isn't really necessary here. If this method returns false then we re-download the file and move it on top of the old one. I just added the delete defensively.